integration/setup.go

```diff
@@ -50,110 +52,115 @@ var (
 )
 
 // SetupOpts represents setup options.
+//
+// TODO add option to use read-only user
 type SetupOpts struct {
+	// If true, returns two client connections to different systems for compatibility test.
+	CompatTest bool
+
 	// Database to use. If empty, temporary test-specific database is created.
 	DatabaseName string
 
 	// Data providers.
 	Providers []shareddata.Provider
 }
 
-// SetupWithOpts setups the test according to given options,
-// and returns test-specific context (that is canceled when the test ends), database collection
-// and the port of the running server.
-func SetupWithOpts(t *testing.T, opts *SetupOpts) (context.Context, *mongo.Collection, int) {
-	t.Helper()
+// SetupResult represents setup results.
+type SetupResult struct {
+	Ctx              context.Context
+	Collection       *mongo.Collection
+	Port             uint16
+	CompatCollection *mongo.Collection
+	CompatPort       uint16
+}
+
+// SetupWithOpts setups the test according to given options.
```

Comment:
I'd say we need to document setup for tests better. I guess, the place to do it is here.
I expect some extended info here. For example, something like "If ports for both target and reference system are set, `ComatCollection` and `CompatPort` are defined as xyz, otherwise they are not set and musnt' be used".

Reply:
That will be in CONTRIBUTING.md in the future PR

Reply:
Ok(ish)!